### PR TITLE
Car and Engine classes finished, method clone added to each

### PR DESCRIPTION
**Pull request by NikolaiKatsai**

### CODE REVIEW

File: src/main/java/core/basesyntax/Engine.java [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+    Engine(String type, int horsePower, int mileage, int weight, int age) {
+        this.type = type;
+        this.horsePower = horsePower;
+        this.mileage = mileage;
+        this.weight = weight;
+        this.age = age;
+    }
+
+    public Engine clone() {
+        return new Engine(type, horsePower, mileage, weight, age);
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Engine.java]: @NikolayKatsai why you are not using new String(type) as in Car class?

Comment (boroda4436) [reply src/main/java/core/basesyntax/Engine.java]: @NikolayKatsai check the both cases: new String(type) and type. Clone the object. Change the type param. Will you see any difference?

Comment (NikolayKatsai) [reply src/main/java/core/basesyntax/Engine.java]: Yep, sorry, forgot to fix it in Engine class))

File: src/main/java/core/basesyntax/Car.java [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
+    public Car(String name, String registrationNumber, String color, Engine engine, int age) {
+        this.name = name;
+        this.registrationNumber = registrationNumber;
+        this.color = color;
+        this.engine = engine;
+        this.age = age;
+    }
+
+    public Car clone() {
+        return new Car(new String(name), new String(registrationNumber),
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Car.java]: @NikolayKatsai String is already immutable. You don't need to to use new String

**Pull request closed without merging**